modules/ppcp-googlepay/src/GooglePayGateway.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * The Google Pay Payment Gateway
+ *
+ * @package WooCommerce\PayPalCommerce\Googlepay
+ */
+
+declare(strict_types=1);
+
+namespace WooCommerce\PayPalCommerce\Googlepay;
+
+use Exception;
+use WC_Order;
+use WC_Payment_Gateway;
+use WooCommerce\PayPalCommerce\ApiClient\Exception\PayPalApiException;
+use WooCommerce\PayPalCommerce\Session\SessionHandler;
+use WooCommerce\PayPalCommerce\WcGateway\Exception\GatewayGenericException;
+use WooCommerce\PayPalCommerce\WcGateway\Exception\PayPalOrderMissingException;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\Messages;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\ProcessPaymentTrait;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\TransactionUrlProvider;
+use WooCommerce\PayPalCommerce\WcGateway\Processor\OrderProcessor;
+use WooCommerce\PayPalCommerce\WcGateway\Processor\RefundProcessor;
+
+/**
+ * Class GooglePayGateway
+ */
+class GooglePayGateway extends WC_Payment_Gateway {
+	use ProcessPaymentTrait;
+
+	const ID = 'ppcp-googlepay';
+
+	/**
+	 * The processor for orders.
+	 *
+	 * @var OrderProcessor
+	 */
+	protected $order_processor;
+
+	/**
+	 * The function return the PayPal checkout URL for the given order ID.
+	 *
+	 * @var callable(string):string
+	 */
+	private $paypal_checkout_url_factory;
+
+	/**
+	 * The Refund Processor.
+	 *
+	 * @var RefundProcessor
+	 */
+	private $refund_processor;
+
+	/**
+	 * Service able to provide transaction url for an order.
+	 *
+	 * @var TransactionUrlProvider
+	 */
+	protected $transaction_url_provider;
+
+	/**
+	 * The Session Handler.
+	 *
+	 * @var SessionHandler
+	 */
+	protected $session_handler;
+
+	/**
+	 * GooglePayGateway constructor.
+	 *
+	 * @param OrderProcessor          $order_processor The Order Processor.
+	 * @param callable(string):string $paypal_checkout_url_factory The function return the PayPal checkout URL for the given order ID.
+	 * @param RefundProcessor         $refund_processor The Refund Processor.
+	 * @param TransactionUrlProvider  $transaction_url_provider Service providing transaction view URL based on order.
+	 * @param SessionHandler          $session_handler The Session Handler.
+	 */
+	public function __construct(
+		OrderProcessor $order_processor,
+		callable $paypal_checkout_url_factory,
+		RefundProcessor $refund_processor,
+		TransactionUrlProvider $transaction_url_provider,
+		SessionHandler $session_handler
+	) {
+		$this->id = self::ID;
+
+		$this->method_title       = __( 'Google Pay', 'woocommerce-paypal-payments' );
+		$this->method_description = __( 'Google Pay', 'woocommerce-paypal-payments' );
```

Comment:
The description could be more verbose, like "Pay with Google" or "Continue with Google Pay"